tinygrad/runtime/ops_metal.py

```diff
@@ -70,6 +70,10 @@ def __init__(self, name:str, prg:str, binary:bool=False):
       self.library = unwrap(METAL.device.newLibraryWithData_error_(data, None))
     else:
       options = Metal.MTLCompileOptions.alloc().init()
+      # With fast math enabled, the compiler does reciprocal division optimizations (e.g. x/4 => x * 0.25). 
+      # This causes inaccuracies (e.g. `Tensor(3) + 0.5)/3.5 != 0`).
+      # For an example see `test_resize_upsample_sizes_nearest_cpu` in the onnx test suite
+      options.setFastMathEnabled_(False)
```

Comment:
this can't be good for perf

Reply:
Yeah this may be the wrong trade off. In LLVM it looks like [there is a flag](https://llvm.org/docs/LangRef.html#fast-math-flags) to only disable reciprocal optimizations.  In Metal it appears to be just the all or nothing `fast-math` flag.

Reply:
is there a way to disable only the problem flags and keep the other ones enabled?

Reply:
Did some more searching and seems to be a known issue in Metal that you can't selectively disable flags. Interesting discussion here:

https://github.com/gpuweb/gpuweb/issues/2076

I have absolutely no idea if this case actually matters in practical applications. I am fairly certain that it won't be possible to get `test_resize_upsample_sizes_nearest_cpu` pass without this or some sort of major hack.  The imprecise division causes it to round in the wrong direction during the resize operation.

Here's a thread from pytorch on an adjacent issue. Shows a practical example of how model output varies due to fast math inaccuracies: https://github.com/pytorch/pytorch/issues/84936

Reply:
what do you mean by "selectively" disable flags? are you trying to disable flags on an instruction level rather than whole kernel? if so, factor out the computation into it's own function and tag it with `__attribute__((optimize("-fno-fast-math")))` alternatively, there's also `metal::precise` and `metal::fast`